### PR TITLE
Paper- Brittney Payne

### PR DESCRIPTION
**Pull request by brpayne**

### CODE REVIEW

File: scratch.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+
```
Comment (CheezItMan) [inline scratch.py:1]: You don't need to check `scratch.py` into git.  Or you can remove it from git with `git rm scratch.py` and the commit it.

File: swap_meet/clothing.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+from .item import Item
+
+
+class Clothing(Item):
+    def __init__(self, condition = None, age = None):
+        category = "Clothing"
+        super().__init__(category, condition, age)
```
Comment (CheezItMan) [inline swap_meet/clothing.py:7]: You could just do this:

```suggestion
        super().__init__(category= "Clothing", condition, age)
```

File: swap_meet/decor.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+from .item import Item
+
+
+
+class Decor(Item):
```
Comment (CheezItMan) [inline swap_meet/decor.py:5]: 👍

File: swap_meet/item.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
+
+
+
+
```
Comment (CheezItMan) [inline swap_meet/item.py:4]:
```suggestion
```

File: swap_meet/item.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+class Item:
```
Comment (CheezItMan) [inline swap_meet/item.py:5]: 👍

File: swap_meet/vendor.py [hunk 6]
```diff
@@ -0,0 +1,128 @@
+from .item import Item
+
+
+class Vendor: 
+    """
+    Define a class that can keep inventory 
+    """
+    def __init__(self, inventory = None):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:8]: 👍

File: swap_meet/vendor.py [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+        if inventory == None:
+                self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:14]: 👍

File: swap_meet/vendor.py [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+        """
+        Takes in an item and adds it to the Vendor objects inventory attribute
+        """
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:21]: 👍

File: swap_meet/vendor.py [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+        """
+        Takes in an item and removes it from the Vendor objects invntry attribute
+        """
+        if item in self.inventory:
+            self.inventory.remove(item)
+        else:
+            return False
+        return item 
+    
+    def get_by_category(self, category):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:31]: 👍

File: swap_meet/vendor.py [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+        Takes in a preferred category and returns all items in vendor 
+        inventory that match that category
+        """
+        item_by_category = []
+        for item in self.inventory:
+            if item.category == category:
+                item_by_category.append(item)
+        return item_by_category
+
+    def swap_items(self, vendor, item_a, item_b):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:42]: 👍 Very nicely written

File: swap_meet/vendor.py [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+        Takes in an instance of vendor and compares inventory lists, 
+        swaps first item of each list if the list contains >= 1 item
+        """
+        if len(self.inventory) >= 1 and len(vendor.inventory)>= 1:
+            self_first_item = self.inventory[0]
+            vendor_first_item = vendor.inventory[0]
+        else:
+            return False
+        self.swap_items(vendor, self_first_item, vendor_first_item)
+        return True
```
Comment (CheezItMan) [inline swap_meet/vendor.py:68]: This could be a bit simpler.

```suggestion
        if self.inventory and vendor.inventory:
            self_first_item = self.inventory[0]
            vendor_first_item = vendor.inventory[0]
          self.swap_items(vendor, self_first_item, vendor_first_item)
          return True

        return False
```

File: swap_meet/vendor.py [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+        """
+        Takes in a category and iterates over item instances in 
+        self.inventory, returns item in best condition 
+        """
+        best_condition = 0
+        best_item = None
+        for item in self.inventory:
+            if item.category == category:
+                if item.condition >= best_condition:
+                    best_condition += item.condition
```
Comment (CheezItMan) [inline swap_meet/vendor.py:80]: why are you adding here?  `+=`???  I think this passes the tests because of the order in which things appear, but this woudn't work in all cases.

File: swap_meet/vendor.py [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+        swaps items from respective inv. lists if they are present
+        """
+        self_wants = other.get_best_by_category(my_priority)
+
+        vendor_wants = self.get_best_by_category(their_priority)
+        if vendor_wants == None or self_wants == None:
+            return False
+        else:
+            self.swap_items(other, vendor_wants, self_wants)
+            return True
```
Comment (CheezItMan) [inline swap_meet/vendor.py:97]: The else isn't needed

```suggestion
        self.swap_items(other, vendor_wants, self_wants)
        return True
```

File: swap_meet/vendor.py [hunk 14]
```diff
@@ -0,0 +1,128 @@
...
+
+    def get_newest(self):
+        """
+        iterates over item objects in self.inventory and returns item 
+        with the youngest age in (int)
+        """
+        best_age = 999999
+        best_item = None
```
Comment (CheezItMan) [inline swap_meet/vendor.py:105]: best age isn't really the best variable name, maybe smallest age.

File: swap_meet/vendor.py [hunk 15]
```diff
@@ -0,0 +1,128 @@
...
+        """
+        Makes use of the get newest funct for self and other, then 
+        swaps the return items if they are present. 
+        """
+        self_wants = other.get_newest()
+        vendor_wants = self.get_newest()
+        if self_wants == None or vendor_wants == None:
+            return False
+        else:
+            self.swap_items(other, vendor_wants, self_wants)
```
Comment (CheezItMan) [inline swap_meet/vendor.py:124]:
```suggestion
        self.swap_items(other, vendor_wants, self_wants)
```

File: swap_meet/vendor.py [hunk 16]
```diff
@@ -0,0 +1,128 @@
...
+        best_item = None
+        for item in self.inventory:
+            if item.age == None:
+                return False
+            if item.age < best_age:
+                best_age = item.age
+                best_item = item
+        return best_item
+
+    def swap_best_by_age(self, other):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:114]: 👍

File: tests/test_wave_06.py [hunk 17]
```diff
@@ -129,3 +129,46 @@ def test_swap_best_by_category_no_other_match():
     assert item_a in tai.inventory
     assert item_b in tai.inventory
     assert item_c in tai.inventory
+
+
+
+def test_get_newest():
```
Comment (CheezItMan) [inline tests/test_wave_06.py:135]: You should also test the `get_newest()` when one or another list is empty.

You should also write a test for what happens when two items are the same age.

File: tests/test_wave_06.py [hunk 18]
```diff
@@ -129,3 +129,46 @@ def test_swap_best_by_category_no_other_match():
...
+    tai = Vendor(
+        inventory=[item_a, item_b, item_c, item_d, item_e]
+    )
+
+    best_item = tai.get_newest()
+
+    assert best_item.age == 2
+
+
+def test_swap_best_by_age():
```
Comment (CheezItMan) [inline tests/test_wave_06.py:150]: Great tests btw.

Suggestion:

1.  Test when each `Vendor` has an empty list of items.
1.  Test when a vendor has multiple items of the smallest age.

**Review by CheezItMan — COMMENTED** [review]
Well done Brittney, you hit the learning goals here.  Well done.  I had a few comments, almost all on minor issues of coding style.  

This was a great submission and you even got the optionals with good tests as well.
